txcache/interface.go

```diff
@@ -16,7 +15,7 @@ type MempoolHost interface {
 
 // SelectionSession provides blockchain information for transaction selection
 type SelectionSession interface {
-	GetAccountState(accountKey []byte) (state.UserAccountHandler, error)
+	GetAccountNonceAndBalance(accountKey []byte) (uint64, *big.Int, bool, error)
```

Comment:
Now `TxCache` only sees what's needed. And we'll nicely get the nonce and balance for missing accounts, as well (no need to code that logic within the _txcache_ package).